community/kernel/src/main/java/org/neo4j/kernel/NeoStoreDataSource.java

```diff
@@ -876,19 +874,21 @@ public synchronized void stop()
             awaitAllTransactionsClosed();
 
             // Write new checkpoint in the log only if the kernel is healthy.
-            // We cannot throw here since we need to shutdown without exceptions.
-            if ( databaseHealth.isHealthy() )
+            // We cannot throw here since we need to shutdown without exceptions,
+            // so let's make the checkpointing part of the life, so LifeSupport can handle exceptions properly
+            life.add( new LifecycleAdapter()
```

Comment:
By adding it to life "last" it should be shutdown "first", that order is reversed during stop/shutdown isn't it?